integ/lib/testing-tier.ts

```diff
@@ -0,0 +1,268 @@
+import { Asset } from '@aws-cdk/aws-s3-assets';
+import { CfnOutput, Construct, Duration, Stack, StackProps } from '@aws-cdk/core';
+import { X509CertificatePem, MongoDbInstaller, MongoDbSsplLicenseAcceptance, MongoDbVersion } from 'aws-rfdk';
+import { IWorkerFleet, RenderQueue, Repository } from 'aws-rfdk/deadline';
+import { IRenderFarmDb } from './storage-struct';
+
+// Interface for choosing userData installation options
+interface UserDataConfigProps {
+  installDeadlineClient: boolean;
+  fetchDocdbCert: boolean;
+  testingScriptPath: string;
+}
+
+// Params object for TestingTier
+export interface TestingTierProps extends StackProps {
+  integStackTag: string;
+}
+
+// Class constructor
+export class TestingTier extends Stack {
+
+  private testInstance: BastionHostLinux;
+  private deadlineVersion: string = process.env.DEADLINE_VERSION!.toString();
+  private stagePath: string = process.env.DEADLINE_STAGING_PATH!.toString();
+
+  constructor(scope: Construct, id: string, props: TestingTierProps) {
+    super(scope, id, props);
+
+    // Collect environment variables
+    const infrastructureStackName = 'RFDKIntegInfrastructure' + props.integStackTag;
+
+    // Vpc.fromLookup acquires vpc deployed to the _infrastructure stack
+    const vpc = Vpc.fromLookup(this, 'Vpc', { tags: { StackName: infrastructureStackName }}) as Vpc;
+
+    // Create an instance that can be used for testing; SSM commands are communicated to the
+    // host instance to run test scripts installed during setup of the instance
+    const testInstance: BastionHostLinux = new BastionHostLinux(this, 'Bastion', {
+      vpc,
+      instanceType: new InstanceType('t3.small'),
+      subnetSelection: {
+        subnetType: SubnetType.PRIVATE,
+      },
+    });
+    this.testInstance = testInstance;
+
+    // Output bastion id for use in tests
+    new CfnOutput(this, 'bastionId', {
+      value: testInstance.instanceId,
+    });
+
+  }
+
+  // Grants the bastion permissions to read the renderQueue cert and creates a stack output for its secretARN
+  public configureCert(testSuiteId:string, cert?:X509CertificatePem) {
+    if(cert) {
+      cert.cert.grantRead(this.testInstance);
+      new CfnOutput(this, 'certSecretARN' + testSuiteId, {
+        value: cert.cert.secretArn,
+      });
+    };
+  }
+
+  // Allows the bastion to connect to the docDB/mongoDB instance and creates a stack output for the secretARN for the database
+  public configureDatabase(testSuiteId:string, database:IRenderFarmDb) {
+    const db = database.db;
+    const dbSecret = database.secret!;
+
+    this.testInstance.connections.allowTo(db, Port.tcp(27017));
+    dbSecret.grantRead(this.testInstance);
+
+    new CfnOutput(this, 'secretARN' + testSuiteId, {
+      value: dbSecret.secretArn,
+    });
+  }
+
+  // Allows the bastion to connect to the repository's EFS file system
+  configureEfs(efs:FileSystem) {
+    this.testInstance.connections.allowToDefaultPort(efs);
+  }
```

Comment:
Since this is only called once by the `RepositoryTestingTier` and it only contains a line of code, I wonder if it should just be inlined instead. I believe it would make the calling code more readable/maintainable. What do you think?

Reply:
Yeah, probably, I don't like it when too many functions clog up the code. Will fix.

Reply:
Looks like this one is still around

Reply:
Fixed, thanks for catching.